api_app/db/repositories/resources_history.py

```diff
@@ -0,0 +1,46 @@
+from typing import List
+import uuid
+from azure.cosmos.aio import CosmosClient
+from db.repositories.base import BaseRepository
+from core import config
+from models.domain.resource import Resource, NewResourceHistoryItem
+from pydantic import parse_obj_as
+
+
+class ResourceHistoryRepository(BaseRepository):
+    @classmethod
+    async def create(cls, client: CosmosClient):
+        cls = ResourceHistoryRepository()
+        await super().create(client, config.STATE_STORE_RESOURCES_HISTORY_CONTAINER)
+        return cls
+
+    @staticmethod
+    def resource_history_query(resourceId: str):
+        return f'SELECT * FROM c WHERE c.resourceId = "{resourceId}"'
```

Comment:
do we need this method?
since we know the ID, it's better to not use SQL, and just read it directly -
it's cheaper in terms of RU.

Reply:
I understand what you are saying, but here it's different. we are doing the query by **resourceId** which means it will potentially bring a list of items, since these are all the history items that are related to this resource, each will have a different **id**. The thing you commented on is not the function but strictly a  const for the sql query which is necessary.

Reply:
please make sure. at least somewhere along the process, this resource id is sanitized.

Reply:
oh, I see, but if you query based on the resource ID, and the partition key is the ID -  shouldn't we choose the resource ID to be the partition key? this way the history items for a resource will be stored on the same partition, which will be better in terms of cost and performance

Reply:
I agree. changing the partition key to resourceID.
Just documenting here that even though the operation repo acts in a similar manner it has a partition key of id, most likely because no one noticed.